tests/operator_test.go

```diff
@@ -728,31 +793,38 @@ var _ = Describe("ALL Operator tests", func() {
 					Skip("This test is depends on prometheus infra being available")
 				}
 
-				Eventually(func() bool {
-					var result map[string]interface{}
-					resp := tests.MakePrometheusHTTPRequest(f, "query?query=kubevirt_cdi_cr_ready")
-					defer resp.Body.Close()
-					bodyBytes, err := ioutil.ReadAll(resp.Body)
-					if err != nil {
-						return false
-					}
-					err = json.Unmarshal(bodyBytes, &result)
-					if err != nil {
-						return false
-					}
-					if len(result["data"].(map[string]interface{})["result"].([]interface{})) == 0 {
-						return false
-					}
-					values := result["data"].(map[string]interface{})["result"].([]interface{})[0].(map[string]interface{})["value"].([]interface{})
-					for _, v := range values {
-						if s, ok := v.(string); ok {
-							if s == "1" {
-								return true
-							}
-						}
-					}
-					return false
-				}, 1*time.Minute, 1*time.Second).Should(BeTrue())
+				Eventually(func() int {
+					return getMetricValue("kubevirt_cdi_cr_ready")
+				}, 1*time.Minute, 1*time.Second).Should(BeNumerically("==", 1))
```

Comment:
consider add a 'waitForMetricValue' function since it seems thats what you mostly do with getMetricValue